sdks/python/apache_beam/utils/counters.py

```diff
@@ -131,8 +131,9 @@ class Counter(object):
   SUM = cy_combiners.SumInt64Fn()
   MEAN = cy_combiners.MeanInt64Fn()
 
-  # Distribution Accumulator Fn
-  DISTRIBUTION = cy_combiners.DistributionCounterFn()
+  # Dataflow Distribution Accumulator Fn.
+  # TODO(BEAM-4045): Generalize distribution counter if necessary.
+  DATAFLOW_DISTRIBUTION = cy_combiners.DataflowDistributionCounterFn()
```

Comment:
<!--new_thread; commit:89d005daa5313826869ebdad0aba047ae95febcb; resolved:0-->
Isn't this using the slow one that we were trying to avoid?

Reply:
Nope, only if cython is not available, this will be the slow one. But I'll also check whether current counter is the slow one before adding counter into worker. That's the reason why there is a field called is_cythonized